src/client.rs

```diff
@@ -20,9 +21,11 @@ use crate::page::Page;
 use crate::playlist::{Playlist, PlaylistRequestBuilder, SinglePlaylistRequestBuilder};
 use crate::track::{SingleTrackRequestBuilder, Track, TrackRequestBuilder};
 use crate::user::{SingleUserRequestBuilder, UserRequestBuilder};
+use crate::{Stream, Future};
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub struct Client {
+    host: String,
```

Comment:
Why have you added the host property here? You don't seem to be using it

Reply:
Fixed that. I think it's nicer delegating to the client the work of prepending the host name to URIs.